python/filereceiver/dsat.py

```diff
@@ -68,14 +68,16 @@ def _watch_file_announcements(self, chunk):
         if header.destination_port != 12:
             return
 
-        timestamp = datetime.datetime.utcfromtimestamp(
-            struct.unpack('<i', chunk[4:8])[0])
+        timestamp = datetime.datetime.fromtimestamp(
+            struct.unpack('<i', chunk[4:8])[0],
+            datetime.timezone.utc,
+        )
         fid = struct.unpack('<I', chunk[8:12])[0]
         # next 12 bytes are for GPS position
         length = struct.unpack('<I', chunk[21:25])[0]
 
         self.log(f'image {fid} announced. Length {length}. '
-                 f'Timestamp {timestamp}')
+                 f'Timestamp {timestamp.replace(tzinfo=None)}')
```

Comment:
Here I think it's fine, and probably preferable, to omit `replace(tzinfo=None)`, since it is only for human-readable logging, and the presence of the timezone makes clearer that this is UTC rather than local time.